src/runtime_src/core/edge/drm/zocl/sched_exec.c

```diff
@@ -815,10 +820,19 @@ configure(struct sched_cmd *cmd)
 		 * Now the zdev->ert is heavily used in configure()
 		 * Need cleanup.
 		 */
-		if (!zdev->ert && get_apt_index(zdev, cu_addr) < 0) {
-			DRM_ERROR("CU address %x is not found in XCLBIN\n",
-				  cfg->data[i]);
-			return 1;
+		if (!zdev->ert) {
+			apt_idx = get_apt_index(zdev, cu_addr);
+			if (apt_idx < 0) {
+				DRM_ERROR("CU address %x is not found in XCLBIN\n",
+					  cfg->data[i]);
+				return 1;
+			}
+			if (is_legacy_intr)
+				irq_id = i;
+			else
+				irq_id = zocl_xclbin_intr_id(zdev, apt_idx);
+			exec->zcu[i].irq = zdev->irq[irq_id];
```

Comment:
Initially I was concerned that irq_id might be out of the zdev->irq[MAX_CU_NUM]. But look deep into the code, since IP_INTERRUPT_ID_MASK is 0xFE and IP_INTERRUPT_ID_SHIFT is 1, irq_id will only be between [0, 127] and our MAX_CU_NUM is 128. So we are good :)
Though, I would suggest to add an extra check here to fail gracefully if we get a wrong irq_id in case we bump up MAX CU NUM or something. But given the reason above, I am OK without that check.

Reply:
Let's refine that in the new version of kds.